blockly-core/core/utils/xml.js

```diff
@@ -17,6 +17,12 @@
  * limitations under the License.
  */
 
+/* global Blockly, goog */
+
+// Ignore "Expected a conditional expression and instead saw an
+// assignment" warnings.
+/* jshint -W084 */
```

Comment:
That argument has some merit, but I would claim that the better step in moving towards (b) would be to actually fix up all instances of this in the file. By adding the ignore, we ensure that we're unlikely to ever actually "fix" these.

If, rather, we don't consider this broken, we should update our general linting config to say that this is ok.